src/daemon/messages.go

```diff
@@ -352,7 +352,14 @@ func (intro *IntroductionMessage) Handle(mc *gnet.MessageContext, daemon interfa
 // Process an event queued by Handle()
 func (intro *IntroductionMessage) Process(d Daemoner) {
 	d.RemoveFromExpectingIntroductions(intro.c.Addr)
-	if !intro.valid {
+	if intro.validationError != nil {
+		if intro.validationError == pex.ErrPeerlistFull {
+			peers := d.RandomExchangeable(d.PexConfig().ReplyCount)
+			givpMsg := NewGivePeersMessage(peers)
+			if err := d.SendMessage(intro.c.Addr, givpMsg); err != nil {
```

Comment:
@stdevStark I think I fixed the lint error , but still I am not quite sure PR is complete. Peer should be disconnected after this and it seems to me that such a thing is not done .

Reply:
... after sending the whole message to the peer , not immediately after invoking the function .